src/crate/client/sqlalchemy/dialect.py

```diff
@@ -165,6 +165,7 @@ class CrateDialect(default.DefaultDialect):
     type_compiler = CrateTypeCompiler
     supports_native_boolean = True
     colspecs = colspecs
+    implicit_returning = True
```

Comment:
Will this still work with older CrateDB versions which don't have `RETURNING` ?

Reply:
Yes, as long as you don't add a column with `server_default=FetchedValue()` which requires RETURNING`, e.g:

```py
class LogOld(Base):
    __tablename__ = "logs_old"
    __table_args__ = {"schema": "doc"}

    ts = Column(DateTime, primary_key=True)
    level = Column(String, primary_key=True)
    message = Column(String, primary_key=True)
```

With the schema:

```sql
create table logs_old (ts timestamp with time zone, level text, message text)
```

Reply:
If `implicit_returning` is `False`, it would never add the RETURNING clause to the INSERT statement.

I guess what you mean is that we should not change behaviour for old versions. ( it will be a different exception).

Reply:
Just wanted to make sure that it doesn't add a `RETURNING` to other statements that previously worked

Could you add a changes entry?

Reply:
> Just wanted to make sure that it doesn't add a `RETURNING` to other statements that previously worked

I will test a bit more.

> Could you add a changes entry?

Sure

Reply:
I found an issue when running one of our test suites, where an INSERT statement would add a RETURNING clause. However cannot reproduce it when the model is used outside of the test suite.

Reply:
Ok, so as soon as one column of the PK has a server side generated value, the SQLAlchemy ORM will append a RETURNING (in order to get the full PK for the model instance).

Example that works at the moment, but would fail with this change:

```py
from sqlalchemy import create_engine, Column, String, DateTime, text
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker


engine = create_engine(
    "crate://",
    connect_args={
        "username": "crate",
        "servers": "localhost:4200",
    },
)

Session = sessionmaker(bind=engine)
Base = declarative_base(bind=Session)


class Event(Base):
    __tablename__ = "consumption_tracking"
    __table_args__ = {"schema": "core"}

    ts = Column("ts", DateTime, primary_key=True, server_default=text("current_timestamp"))
    event = Column("event", String, primary_key=True)


if __name__ == "__main__":
    session = Session()
    o = Event(event="status")
    session.add(o)
    session.commit()
    print(o)
```

Reply:
I am not sure if this is a show stopper, because having the primary key in the model but not in the database is an awful workaround for not being able to use the FetchedValue() for _id. What do you think?

Reply:
Can users override the `implicit_returning` value? If so, I think a (breaking?) changes entry that the impact of the change and how to work around it if necessary should be good enough.

Reply:
Yes, you can override it when creating the engine: `create_engine('crate://',..., implicit_returning=False)`.